### ISSUE TITLE
helm chart: Support K8S 1.8+

### ISSUE THREAD

**Issue by mumoshu** [upmc-enterprises/elasticsearch-operator#141]
I have encountered https://github.com/docker-library/elasticsearch/issues/111 after installing a cluster with helm.

Digging a bit, it turned out that K8S 1.8+ [dropped support for annotation-based init containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) so that the init container wasn't run at all in my env(K8S 1.9).

Rewriting the chart template to use an `initContainers` field rather than an annotation would fix the issue.

**Comment by mumoshu** [comment]
This and #138 seems to share the same cause.

**Comment by mumoshu** [comment]
Let me share what worked for me:

```yaml
initContainers:
      - name: sysctl
        image: busybox
        imagePullPolicy: IfNotPresent
        command: ["sysctl", "-w", "vm.max_map_count=262144"]
        securityContext: { "privileged": true }
containers:
# snip
```

**Comment by mumoshu** [comment]
Turns out you'd better use the startup-script-daemonset approach than initContainers because the latter doesn't survive node restarts. See https://github.com/pires/kubernetes-elasticsearch-cluster/issues/85

**Comment by stevesloka** [comment]
Fixes in #149
